src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/mapper/RoleAssignmentToAccessProfileMapperImpl.java

```diff
@@ -0,0 +1,89 @@
+        List<RoleAssignment> roleAssignments = extractRoleAssignments(filteringResult);
+
+        if (hasGrantTypeExcluded(roleAssignments)) {
+            roleAssignments = filterRoleAssignments(roleAssignments);
+        }
+
+        List<AccessProfile> accessProfiles = new ArrayList<>();
+        for (RoleAssignment roleAssignment : roleAssignments) {
+            for (RoleToAccessProfileDefinition roleToAccessProfile : caseTypeDefinition.getRoleToAccessProfiles()) {
+                if (validRoleAssignments(roleToAccessProfile, roleAssignment)) {
+
+                    List<String> authorisationsList = Arrays.asList(roleToAccessProfile.getAuthorisation().split(","));
+
+                    List<String> roleAssignmentAuthorisation = roleAssignment.getAuthorisations();
+
+                    if (roleAssignmentAuthorisation != null && roleAssignmentAuthorisation.size() > 0) {
+                        for (String authorisation : authorisationsList) {
+                            if (roleAssignmentAuthorisation.contains(authorisation)) {
+                                accessProfiles.add(createAccessProfile(roleAssignment, roleToAccessProfile));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return accessProfiles;
+    }
+
+    private boolean validRoleAssignments(RoleToAccessProfileDefinition roleToAccessProfileDefinition,
+                                         RoleAssignment roleAssignment) {
+        return !roleToAccessProfileDefinition.getDisabled()
+            && roleAssignment.getRoleName().equals(roleToAccessProfileDefinition.getRoleName())
+            && !StringUtils.isEmpty(roleToAccessProfileDefinition.getAuthorisation());
+    }
+
+    private List<RoleAssignment> extractRoleAssignments(RoleAssignmentFilteringResult filteringResult) {
+        return filteringResult.getRoleAssignmentRoleMatchingResults()
+            .stream()
+            .map(pair -> pair.getLeft())
+            .collect(Collectors.toList());
+    }
+
+    private AccessProfile createAccessProfile(RoleAssignment roleAssignment,
+                                              RoleToAccessProfileDefinition roleToAccessProfile) {
+        AccessProfile accessProfile = new AccessProfile();
+
+        accessProfile.setReadOnly(roleToAccessProfile.getReadOnly() || roleAssignment.getReadOnly());
+        accessProfile.setClassification(roleAssignment.getClassification());
+
+        String caseTypeAccessProfiles = roleToAccessProfile.getAccessProfiles();
+        accessProfile.setAccessProfiles(Arrays
+            .asList(caseTypeAccessProfiles.split(",")));
+        return accessProfile;
+    }
+
+    private boolean hasGrantTypeExcluded(List<RoleAssignment> roleAssignments) {
+        return roleAssignments.stream()
+            .filter(roleAssignment -> roleAssignment.getGrantType().equals(GrantType.EXCLUDED.name()))
+            .findFirst().isPresent();
+    }
```

Comment:
move this to RoleAssignmentFilteringResult, that's the whole point of adding these classes, so we can put all sorts of utility methods to them